soups/lib/soups.dart

```diff
@@ -0,0 +1,178 @@
+/*Task 3
+
+Завдання: Суп дня преміум
+Умова:
+
+Користувач вводить кількість днів
+𝑛
+n (не більше 30), протягом яких він записує свій вибір «супу дня». Кількість днів має бути в межах від 1 до 30. Якщо користувач вводить неприпустиму кількість днів, програма має попросити його ввести правильне значення.
+
+Для кожного дня користувач вводить одне або кілька чисел від 1 до 7 (через пробіл), які відповідають певному супу:
+
+1 - «Борщ»
+2 - «Солянка»
+3 - «Розсольник»
+4 - «Грибний суп»
+5 - «Том Ям»
+6 - «Курячий бульйон»
+7 - «Суп-пюре з гарбуза»
+
+Приклад: «1 3» - означає, що на цей день було обрано “Борщ” і “Розсольник”.
+
+Програма повинна:
+
+- Вивести список супів, обраних користувачем для кожного дня.
+
+- Підрахувати, скільки разів був обраний кожен суп за весь період.
+
+- Показати, який суп був обраний найчастіше.
+
+- Треба прорекламувати найчастіше обраний суп
+
+- Треба створити масів меню на кожен день тижня, де будуть використовуватися найбільш популярні супи.
+
+- Далі треба створити масив відвідувачів, вони люблять або один або інший суп та зробити так, щоб ці відвідувачі приходили кожен день та брали суп з меню та ставили оцінку за обслуговування. від 1 до 10 на рандом, але якщо є улюблений суп, то оцінка завжди має бути не менше 6.
+
+ООП не використовуємо, можна використовувати різні структури даних, але обʼєкти створювати не треба
+
+Приклад
+
+*обрали супи дня*
+*підбили статистику*
+Наше меню супів на тиждень згідно з уподобань гурмана:
+пн - Борщ
+вт - Грибний суп
+...
+
+Ось оцінки від наших відвідувачів:
+
+День 1:
+Клієнт Вітя 2,
+Клієнт Коля 3,
+Клієнт Алекс 10
+
+День 2:
+Клієнт Вітя 4,
+Клієнт Коля 7,
+Клієнт Алекс 6
+
+ */
+import 'dart:core';
+import 'dart:io';
+import 'dart:math';
+
+void main() {
+  startGame();
+}
+
+const int userSoupsLimitPerDay = 2;
+const int countDays = 5;
+final List<String> usersList = [];
+final Map<int, String> soupCounter = {};
+Map<String, List<Map<int, String>>> userData = {};
+
+const Map<int, String> soups = {
+  1: '«Борщ»',
+  2: '«Солянка»',
+  3: '«Розсольник»',
+  4: '«Грибний суп»',
+  5: '«Том Ям»',
+  6: '«Курячий бульйон»',
+  7: '«Суп-пюре з гарбуза»',
+};
+
+//Main UserFlow
+void startGame() {
+  var dayCounter = 1;
+  var soupId = 0;
+  String userName = '';
+  String? _userInput = createNewUser();
+  if (_userInput == null) {
```

Comment:
remove. this check needs to be inside function. function needs to have cycle while with all checks and return correct input